io-ballerina/file_string_io.bal

```diff
@@ -173,12 +173,66 @@ public function fileWriteJson(@untainted string path, json content) returns @tai
 # ```
 # + path - The path of the XML file
 # + content - XML content to write
+# + xmlOptions - XML writing options(XML entity type and DOCTYPE)
+# + fileWriteOption - file write option(`OVERWRITE` and `APPEND` are the possible values, and the default value is `OVERWRITE`)
 # + return - The null `()` value when the writing was successful or an `io:Error`
-public function fileWriteXml(@untainted string path, xml content) returns Error? {
-    var byteChannel = openWritableFile(path);
+public function fileWriteXml(@untainted string path, xml content, *XmlWriteOptions xmlOptions, FileWriteOption fileWriteOption = OVERWRITE) returns Error? {
+    WritableByteChannel|Error byteChannel;
+    xml writeContent = xml ``;
+    if (xmlOptions.xmlEntityType == DOCUMENT_ENTITY) {
+        if (fileWriteOption == APPEND) {
+            return error ConfigurationError("The APPEND operation not allowed with DOCUMENT");
```

Comment:
```suggestion
            return error ConfigurationError("The file append operation is not allowed for Document Entity");
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/77/commits/96947ca73cc598ad57975b7931a4ebff0ca199ff